src/senselab/audio/tasks/features_extraction/praat_parselmouth.py

```diff
@@ -1,143 +1,1380 @@
+            "articulation_rate": float("nan"),
+            "phonation_ratio": float("nan"),
+            "pause_rate": float("nan"),
+            "mean_pause_dur": float("nan"),
+        }
+
+
+def extract_pitch_values(snd: Union[parselmouth.Sound, Path, Audio]) -> Dict[str, float]:
+    """Estimate Pitch Range.
+
+    Calculates the mean pitch using a wide range and uses this to shorten the range for future pitch extraction
+    algorithms.
+
+    Args:
+        snd (Union[parselmouth.Sound, Path, Audio]): A Parselmouth Sound object or a file path or an Audio object.
+
+    Returns:
+        dict: A dictionary containing the following keys:
+
+            - pitch_floor (float): The lowest pitch value to use in future pitch extraction algorithms.
+            - pitch_ceiling (float): The highest pitch value to use in future pitch extraction algorithms.
+
+    Notes:
+        Values are taken from: [Standardization of pitch-range settings in voice acoustic analysis](https://doi.org/10.3758/BRM.41.2.318)
+
+        The problem observed with doing a really broad pitch search was the occasional error if F1 was low.
+        So crude outlier detection is used to help with this.
+
+        Important: These values are used within other functions, they are not outputs of the full code.
+
+        Different pitch extraction methods in Praat:
+
+        - Cross-correlation (Praat default) vs auto-correlation pitch extraction:
+        both are used in different functions below.
+        - Cross-correlation is better than auto-correlation at finding period-level variation,
+        such as jitter and shimmer, whereas auto-correlation is better at finding intended intonation contours.
+        - [Discussion on this on a Praat Forum](https://groups.io/g/Praat-Users-List/topic/pitch_detection_ac_vs_cc/78829266?p=,,,20,0,0,0::recentpostdate/sticky,,,20,2,20,78829266,previd=1612369050729515119,nextid=1605568402827788039&previd=1612369050729515119&nextid=1605568402827788039)
+
+    Examples:
+        ```python
+        >>> snd = parselmouth.Sound("path_to_audio.wav")
+        >>> pitch_values(snd)
+        {'pitch_floor': 60, 'pitch_ceiling': 250}
+        ```
+    """
+    try:
+        if not isinstance(snd, parselmouth.Sound):
+            snd = get_sound(snd)
+
+        pitch_wide = snd.to_pitch_ac(time_step=0.005, pitch_floor=50, pitch_ceiling=600)
+        # Other than values above, I'm using default hyperparamters
+        # Details: https://www.fon.hum.uva.nl/praat/manual/Sound__To_Pitch__ac____.html
+
+        # remove outliers from wide pitch search
+        pitch_values = pitch_wide.selected_array["frequency"]
+        pitch_values = pitch_values[pitch_values != 0]
+        pitch_values_Z = (pitch_values - np.mean(pitch_values)) / np.std(pitch_values)
+        pitch_values_filtered = pitch_values[abs(pitch_values_Z) <= 2]
+
+        mean_pitch = np.mean(pitch_values_filtered)
+
+        # TODO: what about children?
```

Comment:
@satra @nickcummins41, for now the audio feats extraction code only takes into account adult males and females. how about including children in the loop? i have commented with some suggested floor and ceiling values for pitch extraction. please, let me know what you think. also, let me know if you think there are more functions that need to be adapted to children's voices.

Reply:
for pitch extraction i think female voice range would cover children generally. so it may depend on what aspect of a female range is being considered. also singing voice may yet be a different consideration.